pkg/jsonpatch/jsonpatch.go

```diff
@@ -10,43 +10,48 @@ import (
 
 var errBadJSONDoc = fmt.Errorf("invalid JSON Document")
 
-type JsonPatchOperation struct {
+type PatchOperation struct {
```

Comment:
It also has info about the type of operation. So we can append `operation`. 
We also use jsonpatch as the module name. So whoever imports this will have to use jsonpatch.JsonPatch() which doesn't look good.